src/stko/_internal/calculators/xtb_calculators.py

```diff
@@ -255,20 +260,23 @@ def __init__(  # noqa: PLR0913
         self._calculate_free_energy = calculate_free_energy
         self._calculate_ip_and_ea = calculate_ip_and_ea
         self._electronic_temperature = str(electronic_temperature)
-        self._solvent = solvent
-        self._solvent_model = solvent_model
-        self._solvent_grid = solvent_grid
         self._charge = str(charge)
         self._num_unpaired_electrons = str(num_unpaired_electrons)
         self._unlimited_memory = unlimited_memory
+        self._solvent = solvent
+        self._solvent_model = solvent_model
+        self._solvent_grid = solvent_grid
+        self._write_sasa_info = write_sasa_info
 
     def _check_path(self, path: Path) -> None:
         if not path.exists():
             msg = f"XTB not found at {path}"
             raise PathError(msg)
 
     def _write_detailed_control(self) -> None:
-        string = f"$gbsa\n   gbsagrid={self._solvent_grid}"
+        string = f"$gbsa\n   gbsagrid={self._solvent_grid}\n"
+        if self._write_sasa_info:
+            string += "$write\n   gbsa=true\n"
```

Comment:
`+=` on a string is a bit of an anti pattern because repeated use of it has very bad performance. While that doesnt apply here because it only happens once, consider

```python
sasa_info = "$write\n gbsa=true\n" if self._write_sasa_info else ""
string = f"$gbsa\n gbsagrid={self._solvent_grid}\n{sasa_info}"
```

Reply:
I did not realise this, updated!